doc/reference/javascript_reference.rst

```diff
@@ -139,71 +151,223 @@ and if necessary, will create/recreate the corresponding bundles.
-  lower level building blocks for the odoo framework.  Note that it contains the
+  lower level building blocks for the odoo framework. Note that it contains the
   *boot.js* file, which defines the odoo module system.
 
-- *web.assets_backend*: this bundle contains the code specific to the web client
-  (notably the web client/action manager/views)
+- *assets_backend*: this bundle contains the code specific to the web client
+  (notably the web client/action manager/views) and all static XML templates used
+  in the backend environment
 
-- *web.assets_frontend*: this bundle is about all that is specific to the public
+- *assets_frontend*: this bundle is about all that is specific to the public
   website: ecommerce, forum, blog, event management, ...
 
+Operations on asset bundles
+---------------------------
 
-Adding files in an asset bundle
--------------------------------
+Typically, handling assets is quite trivial: you just need to add some new files
+to a frequently used bundle like 'common' or 'backend'. But there are other operations
+available to cover use cases bringing additional constraints. Such cases can mostly
+be covered with the following operations.
 
-The proper way to add a file located in *addons/web* to a bundle is simple:
-it is just enough to add a *script* or a *stylesheet* tag to the bundle in the
-file *webclient_templates.xml*.  But when we work in a different addon, we need
-to add a file from that addon.  In that case, it should be done in three steps:
+a) Add one or multiple file(s): append
 
-1. add a *assets.xml* file in the *views/* folder
-2. add the string 'views/assets.xml' in the 'data' key in the manifest file
-3. create an inherited view of the desired bundle, and add the file(s) with an
-   xpath expression. For example,
+The proper way to add a file to a bundle in any addon is simple: it is just enough
+to add a glob path to the bundle in the file *__manifest__.py* like so:
 
-.. code-block:: xml
+.. code-block:: py
+
+    'assets_common': [
+        'my_addon/static/src/js/**/*',
+    ],
+
+By default, adding a simple string to a bundle will append the files matching the
+glob at the end of the bundle.
+
+b) Add one or multiple file(s) at the beginning of the list: prepend
+
+Sometimes you will need to call a certain file before any other in a bundle, when
+loading css file for example. In this case you can use the *prepend* directive
+like so:
+
+.. code-block:: py
+
+    'assets_common': [
+        ('prepend', 'my_addon/static/src/css/bootstrap_overridden.scss'),
+    ],
+
+c) Add one or multiple file(s) before a specofic file: before
+
+Prepending a file at the beginning of a bundle might not be precise enough. The
+*before* directive can then be used to add the resulting files right before the
+target file:
+
+.. code-block:: py
+
+    'assets_common': [
+        ('before', 'web/static/src/css/bootstrap_overridden.scss', 'my_addon/static/src/css/bootstrap_overridden.scss'),
```

Comment:
Do operations like `before` accept "glob" arguments?
Which arguments can be globs and which cannot?

Reply:
For consistency purpose: all manifest paths and targets are parsed as globs. For targets, the new paths are inserted after (or before) the first target path found. This is useless (or at least not used in the current codebase) but it allows to declare all paths in the same form.